### PR TITLE
Shell integrations for units

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8711/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8712/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8713/

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 1]
```diff
@@ -693,6 +693,18 @@ Browser: ${navigator.userAgent}`
...
+    return (
+      this.applicationConfig.flags.terminal ||
+      // Always allow for opening the terminal if the user specified a
+      // jujushell URL in the GUI settings.
+      !!localStorage.getItem('jujushell-url') ||
+      // Also allow for opening the terminal if the user deployed the juju
+      // shell charm.
+      !!this.db.environment.get('jujushellURL')
+    );
+  }
```
Comment (jaycee) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: :+1: thanks!

**Comment by jaycee** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 2]
```diff
@@ -693,6 +693,18 @@ Browser: ${navigator.userAgent}`
     this._clearDeployment(state, noop);
     this._clearInspector(state, noop);
   }
+
+  _showSSHButtons() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: docbloc plz

**Pull request merged**